test/cpplitetests.cpp

```diff
@@ -0,0 +1,510 @@
+
+        tmp_dir = "/tmp/blobfuseteststmp";
+        errno = 0;
+        struct stat buf;
+        int statret = stat(tmp_dir.c_str(), &buf);
+        if (statret == 0)
+        {
+            errno = 0;
+            destroy_path(tmp_dir);
+            ASSERT_EQ(0, errno) << "SetUp - cleanup of old tmp directory failed with errno " << errno;
+        }
+        errno = 0;
+        mkdir(tmp_dir.c_str(), 0777);
+        ASSERT_EQ(0, errno) << "SetUp - tmp dir creation failed with errno " << errno;
+    }
+
+    virtual void TearDown()
+    {
+        test_blob_client_wrapper->delete_container(container_name);
+        destroy_path(tmp_dir);
+    }
+    void run_upload_download(size_t file_size);
+};
+
+std::shared_ptr<blob_client_wrapper> BlobClientWrapperTest::test_blob_client_wrapper = NULL;
+
+// Helper method to follow continuation tokens and list all blobs.  C&P from blobfuse code, we should probably move it to cpplite, although not until we figure out proper retries.
+std::vector<list_blobs_hierarchical_item> list_all_blobs(std::string container, std::string delimiter, std::string prefix)
+{
+    std::vector<list_blobs_hierarchical_item> results;
+
+    std::string continuation;
+
+    std::string prior;
+    do
+    {
+        errno = 0;
+        list_blobs_hierarchical_response response = BlobClientWrapperTest::test_blob_client_wrapper->list_blobs_hierarchical(container, delimiter, continuation, prefix);
+        if (errno == 0)
+        {
+            continuation = response.next_marker;
+            if(response.blobs.size() > 0)
+            {
+                auto begin = response.blobs.begin();
+                if(response.blobs[0].name == prior)
+                {
+                    std::advance(begin, 1);
+                }
+                results.insert(results.end(), begin, response.blobs.end());
+                prior = response.blobs.back().name;
+            }
+        }
+        else
+        {
+            return results; // errno should be set already.
+        }
+    } while (continuation.size() > 0);
+
+    return results;
+}
+
+
+
+void write_to_file(std::string path, std::string text)
+{
+    std::ofstream output_stream(path, std::ios::binary);
+    output_stream << text;
+}
+
+void read_from_file(std::string path, std::string &text)
+{
+    std::ifstream input_stream(path, std::ios::binary);
+    std::stringstream string_stream;
+    string_stream << input_stream.rdbuf();
+    text =  string_stream.str();
+}
+
+// Test to ensure that we can round-trip a small blob, and see it in a listing operation.
+TEST_F(BlobClientWrapperTest, BlobPutDownload)
+{
+    errno = 0;
+    std::vector<list_blobs_hierarchical_item> blobs = list_all_blobs(container_name, "/", "");
+    ASSERT_EQ(0, errno);
+    ASSERT_EQ(0, blobs.size());
+
+    // Create a file
+    std::string file_path = tmp_dir + "/tmpfile";
+    std::string file_text = "some file text here.";
```

Comment:
We are not testing parallel upload/download because the file is smaller than 16MB. Any chance we create a random 100MB file or something and validate that instead of this ? Validation can use md5 sum

Reply:
Ok just saw you are testing large files as well.. Can we do a checksum on that to compare the actual data rather than the file size ?

Reply:
The more-extensive tests below (run_upload_download) do validate the content via RNG